Core/src/main/java/put/io/black/java/core/rest/ScenarioController.java

```diff
@@ -88,28 +230,95 @@ public String getScenarioSteps(@RequestBody String body) {
      */
     @RequestMapping(value = "/nesting", method = RequestMethod.POST, produces = "application/json")
     public String getScenarioNesting(@RequestBody String body) {
+        // Logs
+        logger.info("POST count scenario nesting");
         logger.debug(body);
-        ScenarioManager scenarioManager = new ScenarioManager(body);
-        return Integer.toString(scenarioManager.countScenarioNesting());
+
+        boolean success = false;
+        Integer result = 0;
+
+        // Parser
+        JsonElement scenarioElement = new JsonParser().parse(body).getAsJsonObject().get("scenario");
+        if(scenarioElement != null) {
+            String scenario = scenarioElement.getAsString();
+
+            // Log scenario
+            logger.debug(scenario);
+
+            // Calculate
+            ScenarioManager scenarioManager = new ScenarioManager(scenario);
+            result = scenarioManager.countScenarioNesting();
+
+            success = true;
+        }
+
+        // Response
+        JsonObject response = new JsonObject();
+        if(success) {
+            response.addProperty("status", "success");
+            response.addProperty("result", result);
+        } else {
+            response.addProperty("status", "error");
+            response.addProperty("message", "Missing scenario field in body.");
+        }
+        return response.toString();
     }
 
     /**
      * Scenario to limit nesting level
      * @param body Scenario text
-     * @param toLevel Nesting level limit
      * @return Scenario with limit of nesting level
      */
     @RequestMapping(value = "/level", method = RequestMethod.POST, produces = "application/json")
-    public String getScenarioToLevel(@RequestBody String body, @RequestParam(value = "level", defaultValue = "1") String toLevel) {
+    public String getScenarioToLevel(@RequestBody String body) {
+        // Logs
+        logger.info("POST scenario to nesting level");
         logger.debug(body);
-        logger.debug(toLevel);
+
+        boolean success = false;
+        boolean correctLevel = false;
         int level = 0;
-        try {
-            level = Integer.parseInt(toLevel);
-        } catch (NumberFormatException e) {
-            return ERROR_MESSAGE;
+        String result = "";
+
+        // Level
+        JsonElement levelElement = new JsonParser().parse(body).getAsJsonObject().get("level");
+        if(levelElement != null) {
+            // Try cast to int
+            level = levelElement.getAsInt();
```

Comment:
not throw NumberFormatException ?

Reply:
Oh, I forgot about send not number value.